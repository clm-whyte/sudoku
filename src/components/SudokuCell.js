import React from "react";
import Button from "@material-ui/core/Button";
import { StylesProvider } from "@material-ui/core/styles";

export default function SudokuCell(props) {
  return (
    <StylesProvider injectFirst>
      <Button
        variant="outlined"
        aria-label={props.cellID}
        data-test={props.cellID}
        onClick={props.onClick}
        selected={props.selected}
        className={`sudokuCell
        ${props.row % 3 === 0 ? "boxBoundaryTop" : ""}
        ${props.row % 3 === 2 ? "boxBoundaryBottom" : ""}
        ${props.row === 8 ? "boxBoundaryBottom" : ""}
        ${props.col % 3 === 0 ? "boxBoundaryLeft" : ""}
        ${props.col % 3 === 2 ? "boxBoundaryRight" : ""}
        ${props.col === 8 ? "boxBoundaryRight" : ""}
        ${props.selected ? "selected" : ""}
        `}
      >
        {props.value}
      </Button>
    </StylesProvider>
  );
}
