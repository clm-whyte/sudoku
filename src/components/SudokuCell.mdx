import { Story, Preview } from "@storybook/addon-docs/blocks";
import SudokuCell from "./SudokuCell.js";

# Cell

In standard Sudoku, the grid is made up of 81 **cells** in a 9x9 square (9 rows and 9 columns).  
The **cells** are arranged into 3x3 sections called **boxes**, which are outlined in bold.

## Default

By default, a cell is has no value, is unselected and has a thin grey outline.  
The cells are **Material UI** Buttons which have been restyled using some Sassy CSS.

```css
.sudokuCell {
  max-width: 34px;
  max-height: $cellSize;
  min-width: 34px;
  min-height: $cellSize;
  border: $cellBorder;
  border-radius: 0;
  background-color: white;
  color: navy;
}
```

<Preview>
  <Story id="cell--empty" />
</Preview>

## Position

A cell's position in the grid determines its outline.  
If it's in the first row/column, it will have a thick top/left border respectively.  
Otherwise, every 3rd row/column is given a thick bottom/right border, based on the following lines of code:

```javascript
${props.col % 3 === 2 ? "boxBoundaryRight" : ""}
${props.row % 3 === 2 ? "boxBoundaryBottom" : ""}
```

<Preview>
  <Story id="cell--empty" />
  <Story id="cell--outer-corner" />
  <Story id="cell--inner-corner" />
</Preview>

## Selection

By default a cell is unselected.  
An unselected cell has a white background and may or may not be filled with a value.

A selected cell is denoted by a pink background colour.

<Preview>
  <Story id="cell--empty" />
  <Story id="cell--selected" />
</Preview>

## Cursor

There can be only one "cursor" Cell on the grid.  
The cursor shows the current position of the "active" cell. By default, this is at position `0, 0` and unselected.

<Preview>
  <Story id="cell--empty" />
  <Story id="cell--cursor" />
  <Story id="cell--cursor-and-selected" />
</Preview>

## Value

The cell values default to a value of `null`.

<Preview>
  <Story id="cell--empty" />
  <Story id="cell--filled" />
</Preview>
