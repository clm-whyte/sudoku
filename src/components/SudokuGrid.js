import React from "react";
import SudokuCell from "./SudokuCell";
import Grid from "@material-ui/core/Grid";

class SudokuGrid extends React.Component {
  constructor(props) {
    super(props);
    const grid = [];
    for (let i = 0; i < 9; i++) {
      grid.push(Array(9).fill({ value: null, selected: false, cursor: false }));
    }
    this.state = { cells: grid, cursorLocation: { row: 0, col: 0 } };
  }

  deselectAllCells() {
    const cells = this.state.cells.slice();
    for (let i = 0; i < 9; i++) {
      cells[i] = cells[i].map((cell, index) => {
        cell.selected = false;
        cell.cursor = false;
        return cell;
      });
    }
    this.setState({ cells });
  }

  moveCursor(newRow, newCol) {
    const cursorLocation = this.state.cursorLocation;
    cursorLocation.row = newRow;
    cursorLocation.col = newCol;
    this.setState({ cursorLocation });
  }

  handleClick(row, col, e) {
    if (!e.ctrlKey) {
      this.deselectAllCells();
    }
    this.moveCursor(row, col);
    const cells = this.state.cells.slice();
    cells[row][col] = {
      value: cells[row].length * row + col,
      selected: true,
      cursor: true,
    };
    this.setState({ cells });
  }

  renderCell(row, col) {
    return (
      <SudokuCell
        value={this.state.cells[row][col].value}
        selected={this.state.cells[row][col].selected}
        cursor={
          row === this.state.cursorLocation.row &&
          col === this.state.cursorLocation.col
            ? true
            : false
        }
        key={this.state.cells[row].length * row + col}
        cellID={`cell-${row}/${col}`}
        onClick={(e) => this.handleClick(row, col, e)}
        row={row}
        col={col}
      />
    );
  }

  render() {
    let grid = [];
    for (let rowNumber = 0; rowNumber < 9; rowNumber++) {
      let row = [];
      for (let colNumber = 0; colNumber < 9; colNumber++) {
        row.push(this.renderCell(rowNumber, colNumber));
      }
      grid.push(
        <div
          data-test={`row-${rowNumber}`}
          key={"row" + rowNumber}
          className="grid-row"
        >
          {row}
        </div>
      );
    }
    return (
      <Grid className="grid" data-test="grid">
        {grid}
      </Grid>
    );
  }
}

export default SudokuGrid;
