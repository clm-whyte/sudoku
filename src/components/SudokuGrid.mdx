import { Story, Preview } from "@storybook/addon-docs/blocks";
import SudokuGrid from "./SudokuGrid.js";

# Grid

I'm going to try and add a little bit about the default Cell here.

<Story id="grid--empty" />

<Preview>
  <Story id="grid--empty"></Story>
</Preview>
