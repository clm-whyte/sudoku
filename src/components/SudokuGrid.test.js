/// <reference types="jest" />
import React from "react";
import { mount } from "enzyme";
import SudokuGrid from "./SudokuGrid";
import SudokuCell from "./SudokuCell";

function emptyGrid() {
  const grid = [];
  for (let i = 0; i < 9; i++) {
    grid.push(Array(9).fill({ value: null, selected: false, cursor: false }));
  }
  grid[0][0] = { value: null, selected: false, cursor: true };
  return grid;
}

describe("Clicking on the grid", () => {
  it("clicks on multiple cells on the same row one after the other", () => {
    const gridBefore = emptyGrid();
    const gridAfter = gridBefore.slice();
    gridAfter[0] = [
      { value: 0, selected: false, cursor: false },
      { value: 1, selected: false, cursor: false },
      { value: 2, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: 8, selected: true, cursor: true },
    ];

    const sudokuGrid = mount(<SudokuGrid state={gridBefore} />);
    const cell0 = sudokuGrid.find(SudokuCell).at(0);
    const cell1 = sudokuGrid.find(SudokuCell).at(1);
    const cell2 = sudokuGrid.find(SudokuCell).at(2);
    const cell8 = sudokuGrid.find(SudokuCell).at(8);

    expect(sudokuGrid.state().cells).toEqual(gridBefore);

    cell0.simulate("click");
    cell1.simulate("click");
    cell2.simulate("click");
    cell8.simulate("click");

    expect(sudokuGrid.state().cells).toEqual(gridAfter);
    sudokuGrid.unmount();
  });

  it("clicks on a cell and then CTRL + Clicks two more", () => {
    const gridBefore = emptyGrid();
    const gridAfter = gridBefore.slice();
    gridAfter[0] = [
      { value: 0, selected: true, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: 8, selected: true, cursor: false },
    ];
    gridAfter[8] = [
      { value: 72, selected: true, cursor: true },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
    ];

    const sudokuGrid = mount(<SudokuGrid state={gridBefore} />);
    const cell0 = sudokuGrid.find(SudokuCell).at(0);
    const cell8 = sudokuGrid.find(SudokuCell).at(8);
    const cell72 = sudokuGrid.find(SudokuCell).at(72);

    expect(sudokuGrid.state().cells).toEqual(gridBefore);

    cell0.simulate("click");
    cell8.simulate("click", { ctrlKey: true });
    cell72.simulate("click", { ctrlKey: true });

    expect(sudokuGrid.state().cells).toEqual(gridAfter);
    sudokuGrid.unmount();
  });

  it("clicks on a cell and then SHIFT + Clicks two more on the same row", () => {
    const gridBefore = emptyGrid();
    const gridZeroToFour = gridBefore.slice();
    gridZeroToFour[0] = [
      { value: 0, selected: true, cursor: true },
      { value: null, selected: true, cursor: false },
      { value: null, selected: true, cursor: false },
      { value: null, selected: true, cursor: false },
      { value: 4, selected: true, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
      { value: null, selected: false, cursor: false },
    ];
    const gridZeroToEight = gridBefore.slice();
    gridZeroToEight[0] = [
      { value: 0, selected: true, cursor: true },
      { value: null, selected: true, cursor: false },
      { value: null, selected: true, cursor: false },
      { value: null, selected: true, cursor: false },
      { value: 4, selected: true, cursor: false },
      { value: null, selected: true, cursor: false },
      { value: null, selected: true, cursor: false },
      { value: null, selected: true, cursor: false },
      { value: 8, selected: true, cursor: false },
    ];

    const sudokuGrid = mount(<SudokuGrid state={gridBefore} />);
    const cell0 = sudokuGrid.find(SudokuCell).at(0);
    const cell4 = sudokuGrid.find(SudokuCell).at(4);
    const cell8 = sudokuGrid.find(SudokuCell).at(8);

    expect(sudokuGrid.state().cells).toEqual(gridBefore);

    cell0.simulate("click");
    cell4.simulate("click", { shiftKey: true });
    expect(sudokuGrid.state().cells).toEqual(gridZeroToFour);

    cell8.simulate("click", { shiftKey: true });
    expect(sudokuGrid.state().cells).toEqual(gridZeroToEight);

    sudokuGrid.unmount();
  });
});
