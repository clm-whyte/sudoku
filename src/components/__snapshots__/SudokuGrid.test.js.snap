// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test shallow render of Sudoku Grid 1`] = `
<WithStyles(ForwardRef(Grid))
  className="grid"
  data-test="grid"
>
  <div
    className="grid-row"
    data-test="row-0"
    key="row0"
  >
    <SudokuCell
      cellID="cell-0/0"
      col={0}
      cursor={true}
      key="0"
      onClick={[Function]}
      row={0}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-0/1"
      col={1}
      cursor={false}
      key="1"
      onClick={[Function]}
      row={0}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-0/2"
      col={2}
      cursor={false}
      key="2"
      onClick={[Function]}
      row={0}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-0/3"
      col={3}
      cursor={false}
      key="3"
      onClick={[Function]}
      row={0}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-0/4"
      col={4}
      cursor={false}
      key="4"
      onClick={[Function]}
      row={0}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-0/5"
      col={5}
      cursor={false}
      key="5"
      onClick={[Function]}
      row={0}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-0/6"
      col={6}
      cursor={false}
      key="6"
      onClick={[Function]}
      row={0}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-0/7"
      col={7}
      cursor={false}
      key="7"
      onClick={[Function]}
      row={0}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-0/8"
      col={8}
      cursor={false}
      key="8"
      onClick={[Function]}
      row={0}
      selected={false}
      value={null}
    />
  </div>
  <div
    className="grid-row"
    data-test="row-1"
    key="row1"
  >
    <SudokuCell
      cellID="cell-1/0"
      col={0}
      cursor={false}
      key="9"
      onClick={[Function]}
      row={1}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-1/1"
      col={1}
      cursor={false}
      key="10"
      onClick={[Function]}
      row={1}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-1/2"
      col={2}
      cursor={false}
      key="11"
      onClick={[Function]}
      row={1}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-1/3"
      col={3}
      cursor={false}
      key="12"
      onClick={[Function]}
      row={1}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-1/4"
      col={4}
      cursor={false}
      key="13"
      onClick={[Function]}
      row={1}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-1/5"
      col={5}
      cursor={false}
      key="14"
      onClick={[Function]}
      row={1}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-1/6"
      col={6}
      cursor={false}
      key="15"
      onClick={[Function]}
      row={1}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-1/7"
      col={7}
      cursor={false}
      key="16"
      onClick={[Function]}
      row={1}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-1/8"
      col={8}
      cursor={false}
      key="17"
      onClick={[Function]}
      row={1}
      selected={false}
      value={null}
    />
  </div>
  <div
    className="grid-row"
    data-test="row-2"
    key="row2"
  >
    <SudokuCell
      cellID="cell-2/0"
      col={0}
      cursor={false}
      key="18"
      onClick={[Function]}
      row={2}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-2/1"
      col={1}
      cursor={false}
      key="19"
      onClick={[Function]}
      row={2}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-2/2"
      col={2}
      cursor={false}
      key="20"
      onClick={[Function]}
      row={2}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-2/3"
      col={3}
      cursor={false}
      key="21"
      onClick={[Function]}
      row={2}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-2/4"
      col={4}
      cursor={false}
      key="22"
      onClick={[Function]}
      row={2}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-2/5"
      col={5}
      cursor={false}
      key="23"
      onClick={[Function]}
      row={2}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-2/6"
      col={6}
      cursor={false}
      key="24"
      onClick={[Function]}
      row={2}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-2/7"
      col={7}
      cursor={false}
      key="25"
      onClick={[Function]}
      row={2}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-2/8"
      col={8}
      cursor={false}
      key="26"
      onClick={[Function]}
      row={2}
      selected={false}
      value={null}
    />
  </div>
  <div
    className="grid-row"
    data-test="row-3"
    key="row3"
  >
    <SudokuCell
      cellID="cell-3/0"
      col={0}
      cursor={false}
      key="27"
      onClick={[Function]}
      row={3}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-3/1"
      col={1}
      cursor={false}
      key="28"
      onClick={[Function]}
      row={3}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-3/2"
      col={2}
      cursor={false}
      key="29"
      onClick={[Function]}
      row={3}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-3/3"
      col={3}
      cursor={false}
      key="30"
      onClick={[Function]}
      row={3}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-3/4"
      col={4}
      cursor={false}
      key="31"
      onClick={[Function]}
      row={3}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-3/5"
      col={5}
      cursor={false}
      key="32"
      onClick={[Function]}
      row={3}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-3/6"
      col={6}
      cursor={false}
      key="33"
      onClick={[Function]}
      row={3}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-3/7"
      col={7}
      cursor={false}
      key="34"
      onClick={[Function]}
      row={3}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-3/8"
      col={8}
      cursor={false}
      key="35"
      onClick={[Function]}
      row={3}
      selected={false}
      value={null}
    />
  </div>
  <div
    className="grid-row"
    data-test="row-4"
    key="row4"
  >
    <SudokuCell
      cellID="cell-4/0"
      col={0}
      cursor={false}
      key="36"
      onClick={[Function]}
      row={4}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-4/1"
      col={1}
      cursor={false}
      key="37"
      onClick={[Function]}
      row={4}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-4/2"
      col={2}
      cursor={false}
      key="38"
      onClick={[Function]}
      row={4}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-4/3"
      col={3}
      cursor={false}
      key="39"
      onClick={[Function]}
      row={4}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-4/4"
      col={4}
      cursor={false}
      key="40"
      onClick={[Function]}
      row={4}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-4/5"
      col={5}
      cursor={false}
      key="41"
      onClick={[Function]}
      row={4}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-4/6"
      col={6}
      cursor={false}
      key="42"
      onClick={[Function]}
      row={4}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-4/7"
      col={7}
      cursor={false}
      key="43"
      onClick={[Function]}
      row={4}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-4/8"
      col={8}
      cursor={false}
      key="44"
      onClick={[Function]}
      row={4}
      selected={false}
      value={null}
    />
  </div>
  <div
    className="grid-row"
    data-test="row-5"
    key="row5"
  >
    <SudokuCell
      cellID="cell-5/0"
      col={0}
      cursor={false}
      key="45"
      onClick={[Function]}
      row={5}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-5/1"
      col={1}
      cursor={false}
      key="46"
      onClick={[Function]}
      row={5}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-5/2"
      col={2}
      cursor={false}
      key="47"
      onClick={[Function]}
      row={5}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-5/3"
      col={3}
      cursor={false}
      key="48"
      onClick={[Function]}
      row={5}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-5/4"
      col={4}
      cursor={false}
      key="49"
      onClick={[Function]}
      row={5}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-5/5"
      col={5}
      cursor={false}
      key="50"
      onClick={[Function]}
      row={5}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-5/6"
      col={6}
      cursor={false}
      key="51"
      onClick={[Function]}
      row={5}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-5/7"
      col={7}
      cursor={false}
      key="52"
      onClick={[Function]}
      row={5}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-5/8"
      col={8}
      cursor={false}
      key="53"
      onClick={[Function]}
      row={5}
      selected={false}
      value={null}
    />
  </div>
  <div
    className="grid-row"
    data-test="row-6"
    key="row6"
  >
    <SudokuCell
      cellID="cell-6/0"
      col={0}
      cursor={false}
      key="54"
      onClick={[Function]}
      row={6}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-6/1"
      col={1}
      cursor={false}
      key="55"
      onClick={[Function]}
      row={6}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-6/2"
      col={2}
      cursor={false}
      key="56"
      onClick={[Function]}
      row={6}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-6/3"
      col={3}
      cursor={false}
      key="57"
      onClick={[Function]}
      row={6}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-6/4"
      col={4}
      cursor={false}
      key="58"
      onClick={[Function]}
      row={6}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-6/5"
      col={5}
      cursor={false}
      key="59"
      onClick={[Function]}
      row={6}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-6/6"
      col={6}
      cursor={false}
      key="60"
      onClick={[Function]}
      row={6}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-6/7"
      col={7}
      cursor={false}
      key="61"
      onClick={[Function]}
      row={6}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-6/8"
      col={8}
      cursor={false}
      key="62"
      onClick={[Function]}
      row={6}
      selected={false}
      value={null}
    />
  </div>
  <div
    className="grid-row"
    data-test="row-7"
    key="row7"
  >
    <SudokuCell
      cellID="cell-7/0"
      col={0}
      cursor={false}
      key="63"
      onClick={[Function]}
      row={7}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-7/1"
      col={1}
      cursor={false}
      key="64"
      onClick={[Function]}
      row={7}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-7/2"
      col={2}
      cursor={false}
      key="65"
      onClick={[Function]}
      row={7}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-7/3"
      col={3}
      cursor={false}
      key="66"
      onClick={[Function]}
      row={7}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-7/4"
      col={4}
      cursor={false}
      key="67"
      onClick={[Function]}
      row={7}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-7/5"
      col={5}
      cursor={false}
      key="68"
      onClick={[Function]}
      row={7}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-7/6"
      col={6}
      cursor={false}
      key="69"
      onClick={[Function]}
      row={7}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-7/7"
      col={7}
      cursor={false}
      key="70"
      onClick={[Function]}
      row={7}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-7/8"
      col={8}
      cursor={false}
      key="71"
      onClick={[Function]}
      row={7}
      selected={false}
      value={null}
    />
  </div>
  <div
    className="grid-row"
    data-test="row-8"
    key="row8"
  >
    <SudokuCell
      cellID="cell-8/0"
      col={0}
      cursor={false}
      key="72"
      onClick={[Function]}
      row={8}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-8/1"
      col={1}
      cursor={false}
      key="73"
      onClick={[Function]}
      row={8}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-8/2"
      col={2}
      cursor={false}
      key="74"
      onClick={[Function]}
      row={8}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-8/3"
      col={3}
      cursor={false}
      key="75"
      onClick={[Function]}
      row={8}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-8/4"
      col={4}
      cursor={false}
      key="76"
      onClick={[Function]}
      row={8}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-8/5"
      col={5}
      cursor={false}
      key="77"
      onClick={[Function]}
      row={8}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-8/6"
      col={6}
      cursor={false}
      key="78"
      onClick={[Function]}
      row={8}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-8/7"
      col={7}
      cursor={false}
      key="79"
      onClick={[Function]}
      row={8}
      selected={false}
      value={null}
    />
    <SudokuCell
      cellID="cell-8/8"
      col={8}
      cursor={false}
      key="80"
      onClick={[Function]}
      row={8}
      selected={false}
      value={null}
    />
  </div>
</WithStyles(ForwardRef(Grid))>
`;
